src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -171,8 +177,13 @@ bool SoundRecorder::setDevice(const std::string& name)
         m_isCapturing = false;
         m_thread.wait();
 
-        // Open the requested capture device for capturing 16 bits mono samples
-        captureDevice = alcCaptureOpenDevice(name.c_str(), m_sampleRate, AL_FORMAT_MONO16, m_sampleRate);
+        // Determine the recording format
+        ALCenum format = AL_FORMAT_MONO16;
+        if (m_channelCount > 1)
```

Comment:
It's a bit strange to initialize it with "mono" first and then possibly overwrite the variable with "stereo". I would have written the equivalent statement:
```
ALCenum format = (m_channelCount == 1) ? AL_FORMAT_MONO16 : AL_FORMAT_STEREO16;
```

However, regarding @eXpl0it3r's comment: what is the format if the channel count is neither 1 nor 2?